core/backend/src/test/standalone/BinaryGeometryStreamWrite.test.ts

```diff
@@ -0,0 +1,160 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { assert, expect } from "chai";
+import { Id64String } from "@itwin/core-bentley";
+import {
+  Code,
+  ElementGeometry,
+  ElementGeometryBuilderParams,
+  ElementGeometryDataEntry,
+  ElementGeometryOpcode,
+  GeometricElementProps,
+  GeometryPartProps,
+  GeometryPrimitive,
+  GeometryStreamIterator,
+  IModel, PhysicalElementProps, SubCategoryAppearance,
+} from "@itwin/core-common";
+import { Arc3d, LineSegment3d, LineString3d, Point3d } from "@itwin/core-geometry";
+import {
+  DictionaryModel, GeometryPart, IModelDb, PhysicalObject, SpatialCategory,
+} from "../../core-backend";
+import { IModelTestUtils } from "../IModelTestUtils";
+
+export function createNewModelAndCategory(rwIModel: IModelDb, parent?: Id64String) {
+  // Create a new physical model.
+  const [, modelId] = IModelTestUtils.createAndInsertPhysicalPartitionAndModel(rwIModel, IModelTestUtils.getUniqueModelCode(rwIModel, "newPhysicalModel"), true, parent);
+
+  // Find or create a SpatialCategory.
+  const dictionary: DictionaryModel = rwIModel.models.getModel<DictionaryModel>(IModel.dictionaryId);
+  const newCategoryCode = IModelTestUtils.getUniqueSpatialCategoryCode(dictionary, "ThisTestSpatialCategory");
+  const category = SpatialCategory.create(rwIModel, IModel.dictionaryId, newCategoryCode.value);
+  const spatialCategoryId = rwIModel.elements.insertElement(category);
+  category.setDefaultAppearance(new SubCategoryAppearance({ color: 0xff0000 }));
+  // const spatialCategoryId: Id64String = SpatialCategory.insert(rwIModel, IModel.dictionaryId, newCategoryCode.value!, new SubCategoryAppearance({ color: 0xff0000 }));
+
+  return { modelId, spatialCategoryId };
+}
+
+describe("BinaryGeometryStreamWriteTest", () => {
```

Comment:
@swwilson-bsi do your new tests pass? Don't they require changes on the native side?

Reply:
Yes, they require native code changes. I am running these tests locally (installnativeplatform.bat). 
This PR is supposed to be a branch off of the 'imodel02' branch.

Reply:
Thanks. I have switched the target branch for your PR to imodel02.